### DISCUSSION TITLE
auto shutdown

### DISCUSSION THREAD

**Discussion by Gotbitsa** [rand256/valetudo#526] [Ideas]
Hi, I've been happily using your app for a while now and it works great, but I am missing the feature of auto-shutdown. 
As the robot already boots itself when the charging station is connected to electricity, a simple wall socket timer would be sufficient to keep it off unless in use. 
For instance if I want it to vacuum once a week, in the current situation it has to be on and connected to wifi all week. A simple shutdown timer would allow it to shut down after work, and a simple wall socket timer would wake it up by powering up the charging station the same day the robot is programmed to work. Now instead of 7 x 24 =168 hours standby it only has to be online for the 15 minutes of work its programmed to do each week.

**Comment by pidator** [comment]
IIRC the shutdown command just shuts down the "linux subsystem" of the robot and not the whole robot like pushing the buttons. Therefore your idea is not possible, but you could possibly use a smart switch for power "optimizations", see https://github.com/rand256/valetudo/issues/393 .

**Comment by Gotbitsa** [comment]
No problem. Thanks for your reply. Am rather late at the party anyway. I would have expected there to at least be some fail safes that could be triggered, for instance if the battery heats up or something the device should auto turn off, right? In any case - 3d printed dock extension it is ;)
